renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -52,9 +58,222 @@ enum class OpenGLTest
   NumTests
 };
 
+enum class PerFragmentQueryType
+{
+  ShaderOut,
+  PostMod,
+  PrimitiveId
+};
+
+GLuint CreatePrimitiveIdProgram(GLuint Program, GLuint Pipeline, GLuint fragShader,
```

Comment:
To fix the way we replaced fragment shaders fro primitive ID, we copied the code from GLReplay::CreateOverlayProgram and fixed it up to work for pixel history. Should we try to refactor this into a common function we can use in both places?

Reply:
Yes I think this function is probably complex enough that it merits being made into a helper, probably in `GLReplay` itself. It looks like if you refactored it to do the delete/create of the new program outside then the function is already fairly factored out.